### DISCUSSION TITLE
C-Get for RTSTRUCT

### DISCUSSION THREAD

**Discussion by mikhail-rozum** [fo-dicom/fo-dicom#1500] [Q&A]
Hello!
I have a DICOM Server (Orthanc) and I need to download structure set dicom file from there(RTSTRUCT modality). I use this code

`
var request = new DicomCGetRequest(studyInstanceId, seriesInstanceId);

client.OnCStoreRequest += request =>
{
    result.Add(request.Dataset);
    return Task.FromResult(new DicomCStoreResponse(request, DicomStatus.Success));
};
await client.AddRequestAsync(request);
await client.SendAsync();
`
But I get an error:
![image](https://user-images.githubusercontent.com/13016376/213984253-26fed354-ea42-430e-9794-325d2912cc6e.png)
How can I get RTSTRUCT dicom file from dicom server?

**Comment by amoerie** [comment]
Hi @mikhail-rozum 

Does ORTHANC support C-GET? It looks like it just returns a failure response.

If that doesn't work, you can also try C-MOVE, which seems more widely supported.
This means you would run a DICOM server listener to accept inbound C-STORE requests, and then you send a C-MOVE request to ORTHANC asking it to send you the files to your DICOM server listener.
